sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java

```diff
@@ -17,177 +17,93 @@
+        return Strings.isNullOrEmpty(value) ? openApiWrapper.getValue(ConfigKeyUtils.path2Key(key)) : value;
     }
     
     @Override
     public List<String> getChildrenKeys(final String key) {
-        return new ArrayList<>(tree.getChildrenKeys(key));
+        return null;
     }
     
     @Override
     public void persist(final String key, final String value) {
-        String apolloKey = convertKey(key);
-        updateKey(apolloKey, value);
-        publishNamespace();
-        tree.refresh(apolloKey, APOLLO_KEY_SEPARATOR);
-    }
-    
-    private void updateKey(final String key, final String value) {
-        OpenItemDTO openItem = new OpenItemDTO();
-        openItem.setKey(key);
-        openItem.setValue(value);
-        openItem.setComment("ShardingSphere create or update config");
-        openItem.setDataChangeCreatedBy(administrator);
-        client.createOrUpdateItem(appId, env, clusterName, namespace, openItem);
-    }
-    
-    private void publishNamespace() {
-        NamespaceReleaseDTO release = new NamespaceReleaseDTO();
-        release.setReleaseTitle("ShardingSphere namespace release");
-        release.setReleaseComment("ShardingSphere namespace release");
-        release.setReleasedBy(administrator);
-        release.setEmergencyPublish(true);
-        client.publishNamespace(appId, env, clusterName, namespace, release);
+        openApiWrapper.persist(ConfigKeyUtils.path2Key(key), value);
     }
     
     @Override
     public void watch(final String key, final DataChangedEventListener dataChangedEventListener) {
-        apolloConfig.addChangeListener(new ConfigChangeListener() {
+        String apolloKey = ConfigKeyUtils.path2Key(key);
+        caches.put(apolloKey, dataChangedEventListener);
+        ConfigChangeListener listener = new ConfigChangeListener() {
+
             @Override
             public void onChange(final ConfigChangeEvent changeEvent) {
-                for (String key : changeEvent.changedKeys()) {
-                    ConfigChange change = changeEvent.getChange(key);
+                for (String changeKey : changeEvent.changedKeys()) {
+                    ConfigChange change = changeEvent.getChange(changeKey);
                     DataChangedEvent.ChangedType changedType = getChangedType(change.getChangeType());
-                    if (DataChangedEvent.ChangedType.IGNORED != changedType) {
-                        dataChangedEventListener.onChange(new DataChangedEvent(deConvertKey(key), change.getNewValue(), changedType));
+                    if (DataChangedEvent.ChangedType.IGNORED == changedType) {
+                        continue;
+                    }
+                    if (caches.get(changeKey) == null) {
+                        continue;
                     }
+                    caches.get(changeKey).onChange(new DataChangedEvent(ConfigKeyUtils.key2Path(changeKey), change.getNewValue(), changedType));
```

Comment:
Is this line too long?

Reply:
ShardingSphere's checkstyle does not warn, so it is in one line.